site/documentation/content/user-guide/coap-adapter.md

```diff
@@ -0,0 +1,231 @@
+<!--more-->
+
+## Device Authentication
+
+The CoAP adapter by default requires clients (devices or gateway components) to authenticate during connection establishment. The adapter supports currently only [PSK](https://tools.ietf.org/html/rfc4279) as part of a DTLS handshake for that purpose. The other variants mentioned in [Securing CoAP](https://tools.ietf.org/html/rfc7252#section-9) may follow in the future.
+
+The adapter tries to authenticate the device using this mechanisms
+
+### PSK
+
+The *identity* provided in the ClientKeyExchange must have the form *auth-id@tenant*, e.g. `sensor1@DEFAULT_TENANT`. The adapter performs the handshake using the credentials `configured Credentials service - (documentation pending)` has one record for the client. The adapter uses the Credentials API's *get* operation to retrieve the credentials on record with the *tenant* and *auth-id* provided by the device in the *identity* and `psk` as the *type* of secret as query parameters.
+
+The examples below refer to devices `4711` and `gw-1` of tenant `DEFAULT_TENANT` using *auth-ids* `sensor1` and `gw1` and corresponding passwords. The example deployment as described in the [Deployment Guides]({{< relref "deployment" >}}) comes pre-configured with the corresponding entities in its device registry component.
+Please refer to the [Credentials API]({{< relref "/api/credentials#standard-credential-types" >}}) for details regarding the different types of secrets.
+
+**NB** There is a subtle difference between the *device identifier* (*device-id*) and the *auth-id* a device uses for authentication. See [Device Identity]({{< relref "/concepts/device-identity.md" >}}) for a discussion of the concepts.
+
+## Message Limits
+
+Before accepting any telemetry or event or command messages, the CoAP adapter verifies that the configured [message limit] ({{< relref "/concepts/resource-limits.md" >}}) is not exceeded. If the limit is exceeded then the incoming message is discarded with the status code `429 Too Many Requests`. 
+
+## Publish Telemetry Data (authenticated Device)
+
+The device is authenticated using PSK.
+
+* URI: `/telemetry`
+* Method: `POST`
+* Type:
+  * `CON`: *at least once* (`1`) QoS levels
+  * `NON`: *at most once* (`0`) QoS levels
+* Request Options:
+  * (required) `content-type`: The type of payload contained in the request body.
+  * (optional) `URI-query: hono-ttd`: The number of seconds the device will wait for the response.
+  * (optional) `URI-query: empty`: If the request doesn't contain payload.
+* Request Body:
+  * (required) Arbitrary payload encoded according to the given content type.
+* Response Options:
+  * (optional) `content-type`: A media type describing the semantics and format of payload contained in the response body. This header will only be present if the response contains a command to be executed by the device which requires input data.
```

Comment:
I wonder how this should work since there (currently?) is no way to map arbitrary content type strings provided by a downstream application to a CoAP content type option value, or is there?

Reply:
That simply requires a IANA registration for the content type :-).

"arbitrary content type strings" as the ""application/vnd.eclipse-hono-empty-notification" don't work. Maybe adding a supported content-type table at the bottom helps here.

Reply:
That would be a start. We will also need to document what we do with *unsupported* content-types, right?

Reply:
"unsupported content-types" from "north-bound"? They are ignored.
Are there other possibilities? Commands? One-Way commands?